blockdag/blocknode.go

```diff
@@ -198,6 +204,22 @@ func (node *blockNode) RelativeAncestor(distance uint64) *blockNode {
 	return node.SelectedAncestor(node.chainHeight - distance)
 }
 
+// isInPast returns true iff this node can be reached from the other in the DAG.
+// The complexity of this method is O(log(|node.futureBlocks|))
+func (node *blockNode) isInPast(other *blockNode) bool {
+	// First, check if this node is a reachability tree ancestor of the
+	// other node
+	thisTreeInterval := node.reachabilityTreeNode.interval
+	otherTreeInterval := other.reachabilityTreeNode.interval
+	if thisTreeInterval.isAncestorOf(&otherTreeInterval) {
+		return true
+	}
+
+	// Otherwise, use previously registered future blocks to complete the
+	// reachability test
+	return node.futureBlocks.isFutureBlock(other)
```

Comment:
isFutureBlock sounds like a synonym for isInFuture, but apparently, that's not the case, so maybe better find a better name (might be easier after futureBlocks are renamed).